domain/src/main/java/org/oppia/domain/oppialogger/analytics/AnalyticsController.kt

```diff
@@ -154,4 +173,79 @@ class AnalyticsController @Inject constructor(
       )
       .build()
   }
+
+  /**
+   * Checks network connectivity of the device.
+   * Saves the [eventLog] to the [eventLogStore] in the absence of it.
+   * Uploads to remote service in the presence of it.
+   */
+  private fun checkNetworkAndManageEventLog(eventLog: EventLog) {
+    when (networkConnectionUtil.getCurrentConnectionStatus()) {
+      NONE -> addEventLog(eventLog)
+      else -> eventLogger.logEvent(eventLog)
```

Comment:
Related to a later comment, but I think it's important to make sure that in both the internet and non-internet cases that the logs are eventually uploaded.

Reply:
We were making sure that the logs are eventually uploaded using the fake implementation of the eventLogger. We can probably use that here as well.

Reply:
@BenHenning @vinitamurthi Can we resolve this as well ?

Reply:
Ah yes sorry, this is part of M3 we can resolve this